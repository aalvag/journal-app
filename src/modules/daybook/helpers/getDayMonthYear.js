const getDayMonthYear = (dateString) => {
  const date = new Date(dateString);
  return {
    day: date.getDate(),
    month: date.toLocaleString("default", { month: "short" }),
    yearDay: `${date.getFullYear()}, ${date.toLocaleString("default", { weekday: "long" })}`,
  };
};

export default getDayMonthYear;
