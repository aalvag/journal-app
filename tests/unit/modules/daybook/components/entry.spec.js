import { shallowMount } from "@vue/test-utils";
import EntryComponent from "@/modules/daybook/components/EntryOne.vue";
import getDayMonthYear from "@/modules/daybook/helpers/getDayMonthYear";

describe("EntryComponent", () => {
  let wrapper;
  const entry = {
    id: "1",
    text: "Sample entry text for test purposes only.",
    date: new Date(2023, 5 - 1, 30),
  };

  const mockRouter = {
    push: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallowMount(EntryComponent, {
      props: { entry },
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
  });

  it("must macth snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("redirects to entry view", () => {
    const entryView = wrapper.find(".entry-container");
    entryView.trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: entry.id },
    });
  });

  it("renders shortText correctly", () => {
    const expectedShortText = entry.text.length > 100 ? `${entry.text.substring(0, 100)}...` : entry.text;
    expect(wrapper.vm.shortText).toBe(expectedShortText);
  });

  it("renders day correctly", () => {
    const { day } = getDayMonthYear(entry.date);
    expect(wrapper.vm.day).toBe(day);
  });

  it("renders month correctly", () => {
    const { month } = getDayMonthYear(entry.date);
    expect(wrapper.vm.month).toBe(month);
  });

  it("renders yearDay correctly", () => {
    const { yearDay } = getDayMonthYear(entry.date);
    expect(wrapper.vm.yearDay).toBe(yearDay);
  });

  it("renders yearMonth correctly", () => {
    const { yearMonth } = getDayMonthYear(entry.date);
    expect(wrapper.vm.yearMonth).toBe(yearMonth);
  });
});
