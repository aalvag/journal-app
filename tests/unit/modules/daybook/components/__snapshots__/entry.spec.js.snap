// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryComponent must macth snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">30</span><span class="mx-1 fs-5">may</span><span class="mx-2 fw-light">2023, martes</span></div>
  <div class="entry-description">Sample entry text for test purposes only.</div>
</div>
`;
